app/mailers/host_mailer.rb

```diff
@@ -6,7 +6,9 @@ class HostMailer < ApplicationMailer
   # sends out a summary email of hosts and their metrics (e.g. how many changes failures etc).
   def summary(options = {})
     raise ::Foreman::Exception.new(N_("Must specify a valid user with email enabled")) unless (user = User.find(options[:user]))
-    hosts = Host::Managed.authorized_as(user, :view_hosts, Host)
+    hosts = User.as user do
+      Host::Managed.authorized_as(user, :view_hosts, Host)
```

Comment:
Since you are in a user scope, you can use `authorized` instead of `authorized_as` here and below:
```suggestion
      Host::Managed.authorized(:view_hosts)
```